src/ussd_nurse.js

```diff
@@ -131,6 +207,196 @@ go.app = function() {
+                        return is
+                        .optin(self.im.user.answers.registrant.id, "msisdn", self.im.user.answers.registrant_msisdn)
+                        .then(function() {
+                            return 'state_faccode';
+                        });
+                    } else {
+                        return 'state_permission_denied';
+                    }
+                }
+            });
+        });
+
+        self.add('state_faccode', function(name) {
+            var owner = self.im.user.answers.operator.id === self.im.user.answers.registrant.id
+                ? 'your' : 'their';
+            var error = $("Sorry, that code is not recognized. Please enter the 6-digit facility code again, e. 535970:");
+            var question = $("Please enter {{owner}} 6-digit facility code:")
+                .context({owner: owner});
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    return self
+                    .validate_nc_clinic_code(self.im, content)
+                    .then(function(facname) {
+                        if (!facname) {
+                            return error;
+                        } else {
+                            self.im.user.answers.registrant.details.nurseconnect = {};
+                            self.im.user.answers.registrant.details.nurseconnect.facname = facname;
+                            self.im.user.answers.registrant.details.nurseconnect.faccode = content;
+
+                            return null;  // vumi expects null or undefined if check passes
+                        }
+                    });
+                },
+                next: 'state_facname'
+            });
+        });
+
+        self.add('state_facname', function(name) {
+            var owner = self.im.user.answers.operator.id === self.im.user.answers.registrant.id
+                ? 'your' : 'their';
+            return new ChoiceState(name, {
+                question: $("Please confirm {{owner}} facility: {{facname}}")
+                    .context({
+                        owner: owner,
+                        facname: self.im.user.answers.registrant.details.nurseconnect.facname
+                    }),
+                choices: [
+                    new Choice('state_save_nursereg', $('Confirm')),
+                    new Choice('state_faccode', $('Not the right facility')),
+                ],
+                next: function(choice) {
+                    return choice.value;
+                }
+            });
+        });
+
+        self.add('state_save_nursereg', function(name) {
+            // Save useful identity info
+            self.im.user.answers.registrant.details.nurseconnect.is_registered = "true";
+
+            // if (self.im.user.answers.operator.id !== self.im.user.answers.registrant.id) {
```

Comment:
If you're going to leave big chunks of commented out code in (I'd rather you didn't) please state the issue it expected to be uncommented in.